### PR TITLE
Type conversions fixes

### PR DESCRIPTION
**Pull request by monidzik**
Couple of improvements in terms of type conversions; mostly adding `static_casts`.

### CODE REVIEW

File: rclcpp/test/test_duration.cpp [hunk 1]
```diff
@@ -47,15 +47,15 @@ TEST(TestDuration, operators) {
   EXPECT_FALSE(young == old);
 
   rclcpp::Duration add = old + young;
-  EXPECT_EQ(add.nanoseconds(), (rcl_duration_value_t)(old.nanoseconds() + young.nanoseconds()));
+  EXPECT_EQ(add.nanoseconds(), (old.nanoseconds() + young.nanoseconds()));
```
Comment (dirk-thomas) [inline rclcpp/test/test_duration.cpp]: Unnecessary parenthesis.

File: rclcpp/src/rclcpp/utilities.cpp [hunk 2]
```diff
@@ -103,9 +103,9 @@ remove_ros_arguments(int argc, char const * const argv[])
     throw exceptions::RCLError(base_exc, "");
   }
 
-  std::vector<std::string> return_arguments(nonros_argc);
+  std::vector<std::string> return_arguments(static_cast<std::size_t>(nonros_argc));
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/utilities.cpp]: I don't think you need `std::` in front of `size_t` - none of the other code uses it.

Same below.

File: rclcpp/src/rclcpp/parameter_client.cpp [hunk 3]
```diff
@@ -152,7 +152,7 @@ AsyncParametersClient::get_parameters(
       auto & pvalues = cb_f.get()->values;
 
       for (auto & pvalue : pvalues) {
-        auto i = &pvalue - &pvalues[0];
+        size_t i = static_cast<size_t>(&pvalue - &pvalues[0]);
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/parameter_client.cpp]: With the explicit static case you should be able to continue to use `auto` - no need to repeat the type.

File: rclcpp/src/rclcpp/duration.cpp [hunk 4]
```diff
@@ -122,15 +123,15 @@ Duration::operator>(const rclcpp::Duration & rhs) const
...
 bounds_check_duration_sum(int64_t lhsns, int64_t rhsns, uint64_t max)
 {
-  auto abs_lhs = (uint64_t)std::abs(lhsns);
-  auto abs_rhs = (uint64_t)std::abs(rhsns);
+  auto abs_lhs = static_cast<uint64_t>(std::abs(lhsns));
+  auto abs_rhs = static_cast<uint64_t>(std::abs(rhsns));
 
-  if (lhsns > 0 && rhsns > 0) {
-    if (abs_lhs + abs_rhs > (uint64_t) max) {
+  if ((lhsns > 0) && (rhsns > 0)) {
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/duration.cpp]: No need for these extra parenthesis. Please remove them.

Same below.

**Comment by wjwwood** [comment]
@monidzik friendly ping

File: rclcpp/test/test_duration.cpp (hunk 1 above)
Comment (monidzik) [reply rclcpp/test/test_duration.cpp]: removed

File: rclcpp/src/rclcpp/utilities.cpp (hunk 2 above)
Comment (monidzik) [reply rclcpp/src/rclcpp/utilities.cpp]: right, removed

File: rclcpp/src/rclcpp/parameter_client.cpp (hunk 3 above)
Comment (monidzik) [reply rclcpp/src/rclcpp/parameter_client.cpp]: sure, done

File: rclcpp/src/rclcpp/duration.cpp (hunk 4 above)
Comment (monidzik) [reply rclcpp/src/rclcpp/duration.cpp]: removed

File: rclcpp/src/rclcpp/duration.cpp [hunk 5]
```diff
@@ -181,8 +182,10 @@ bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
 {
   auto abs_dns = static_cast<uint64_t>(std::abs(dns));
   auto abs_scale = std::abs(scale);
-
-  if (abs_scale > 1.0 && abs_dns > static_cast<uint64_t>(max / abs_scale)) {
+  long double max_ld = static_cast<long double>(max);
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/duration.cpp]: Please avoid the one-time use variable and use the `static_cast` directly in the condition below.

Comment (monidzik) [reply rclcpp/src/rclcpp/duration.cpp]: ok, changed

**Comment by dirk-thomas** [comment]
Thanks for the patch and for iterating on it. Running a set of CI builds:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8715/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4547/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7118/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8629/)

**Pull request merged**